CP4AIOPS-3113 Pass authentication group delimiter to rails

kbrock opened the pull request:
Each authentication system is configurated a little differently. open idc uses a , as the group delimiter

This sets the configuration in apache so it can easily be passed to the individual containers.

Just setting the value in settings.yml is a viable alternative

see also https://github.com/ManageIQ/manageiq/pull/23139

kbrock committed:
Call out the default group delimiter for mellon is a semicolon

MellonMergeEnvVars On brings back all groups for REMOTE_USER_GROUPS
By default, the delimiter is ;, but this can be overridden

This is a no-op, but it highlights that a semicolon is used to join groups.
Liberty (via oidc) is using a comma, but that is not displayed here.
LookupUserGroups specifies a ":"

Fryguy commented on TEMPLATE/etc/httpd/conf.d/manageiq-remote-user-openidc.conf:
```diff
@@ -10,4 +10,5 @@ RequestHeader set X_REMOTE_USER_FIRSTNAME %{OIDC_CLAIM_GIVEN_NAME}e
 RequestHeader set X_REMOTE_USER_LASTNAME  %{OIDC_CLAIM_FAMILY_NAME}e             env=OIDC_CLAIM_FAMILY_NAME
 RequestHeader set X_REMOTE_USER_FULLNAME  %{OIDC_CLAIM_NAME}e                    env=OIDC_CLAIM_NAME
 RequestHeader set X_REMOTE_USER_GROUPS    %{OIDC_CLAIM_GROUPS}e                  env=OIDC_CLAIM_GROUPS
+RequestHeader set X_REMOTE_USER_DELIMITER ,                                      env=OIDC_CLAIM_GROUPS
```
Not sure if this is the right place for this, but `X_REMOTE_USER_DELIMITER` feels like a confusing name, because it sounds like this delimeter is for the users themselves, and not the user's groups?  I'm thinking `X_REMOTE_USER_GROUPS_DELIMITER` and it happens to match name-wise with `X_REMOTE_USER_GROUPS` showing they are related.

Fryguy commented:
Seems the remote-user-saml.conf changes from https://github.com/ManageIQ/manageiq-appliance_console/pull/265/files are missing here?  Additionally, I think the mellon change from here is needed: https://github.com/ManageIQ/manageiq-pods/pull/1155/files

kbrock committed:
Pass authentication group delimiter to rails

Each authentication system is configurated a little differently.
open idc uses a , as the group delimiter

This sets the configuration in apache so it can easily
be passed to the individual containers.

Just setting the value in settings.yml is a viable alternative

kbrock committed:
Pass group delimiter to rails

For multiple groups, LookupUser sets the delimiter to :
rails will then use that to split up the group parameter

Since saml (mod_auth_melon) uses a ;, we need to create a separate file
This will also need to change in rails

kbrock commented:
update:
- pushed missing change (hadn't pushed :( )

update:
- header now called `X_REMOTE_USER_GROUP_DELIMITER` (note `GROUP` is singular)

Fryguy commented:
Backported to `radjabov` in commit 3d9a7713424a8ce35fdac274936c9bfe895d1f64.

```text
commit 3d9a7713424a8ce35fdac274936c9bfe895d1f64
Author: Jason Frey <fryguy9@gmail.com>
Date:   Tue Nov 19 14:53:03 2024 -0500

    Merge pull request #389 from kbrock/CP4AIOPS-3113
    
    CP4AIOPS-3113 Pass authentication group delimiter to rails
    
    (cherry picked from commit ff7875b548205db44fb1e3d383c00e3da9ef9deb)
```
